backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java

```diff
@@ -392,6 +397,12 @@ private void deleteRoleBindingsOfAccessibleResources(List<RepositoryEntity> allo
       ListBranchesRequest.Response listBranchesResponse =
           listBranches(
               ListBranchesRequest.newBuilder().setRepositoryId(request.getRepositoryId()).build());
+
+      CommitPaginationDTO commitPaginationDTO =
```

Comment:
I prefer the delete optimizations to be part of different PR

Reply:
reverted